pkg/state/state_pod_security_policy.go

```diff
@@ -102,6 +104,13 @@ func (s *statePodSecurityPolicy) Sync(customResource interface{}, infoCatalog In
 // Get a map of source kinds that should be watched for the state keyed by the source kind name
 func (s *statePodSecurityPolicy) GetWatchSources() map[string]*source.Kind {
 	wr := make(map[string]*source.Kind)
+	psp := &policyv1beta1.PodSecurityPolicyList{}
+	err := s.client.List(context.TODO(), psp)
+	if err != nil {
+		// We assume it's k8s v1.25 or newer so PodSecurityPolicy is not supported and no need to reconcile them
+		log.V(consts.LogLevelInfo).Info("pod security policy is not available")
```

Comment:
Do we need to add an error message if helm chart is deployed with PSP enabled on k8s 1.25+?

Reply:
IMO, it should be done on CRD validation stage in an admission controller which isn't implemented yet

Reply:
Fair point.
I guess to explicitly deprecate PSP in NicClusterPolicy we also need an admission controller?